velero-plugins/pod/restore.go

```diff
@@ -37,16 +37,22 @@ func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*v
 	if registry == "" {
 		return nil, fmt.Errorf("failed to find restore registry annotation")
 	}
-	ownerRefs, err := common.GetOwnerReferences(input.ItemFromBackup)
-	if err != nil {
-		return nil, err
-	}
-	if len(ownerRefs) > 0 {
-		p.Log.Infof("[pod-restore] skipping restore of pod %s, has owner references", pod.Name)
-		return velero.NewRestoreItemActionExecuteOutput(input.Item).WithoutRestore(), nil
+
+	if input.Restore.Annotations[common.StageAnnotation] != "" || input.Restore.Annotations[common.MigrateRestorePhase1Annotation] != "" {
+		common.ConfigureContainerSleep(pod.Spec.Containers, "infinity")
+		common.ConfigureContainerSleep(pod.Spec.InitContainers, "0")
+	} else {
+		ownerRefs, err := common.GetOwnerReferences(input.ItemFromBackup)
+		if err != nil {
+			return nil, err
+		}
+		if len(ownerRefs) > 0 {
+			p.Log.Infof("[pod-restore] skipping restore of pod %s, has owner references", pod.Name)
+			return velero.NewRestoreItemActionExecuteOutput(input.Item).WithoutRestore(), nil
+		}
+		common.SwapContainerImageRefs(pod.Spec.Containers, backupRegistry, registry, p.Log)
+		common.SwapContainerImageRefs(pod.Spec.InitContainers, backupRegistry, registry, p.Log)
```

Comment:
As long as the container image that's swapped out in ConfigureContainerSleep (currently "registry.access.redhat.com/rhel7") is never an image in the local registry, it doesn't matter whether the SwapContainerImageRefs is in the else block or outside it. If it's possible that any of the images referenced in pod.Spec.Containers or pod.spec.InitContainers are internal images after ConfigureContainerSleep, then these calls should be moved to outside the else block, since these function calls replace image references into the src cluster's internal registry with references into the destination cluster's internal registry. If the image will always be external, it doesn't matter since SwapContainerImageRefs doesn't do anything if the image in question doesn't begin with the backupRegistry host:port.

Reply:
Sounds like there's no problem with it running so I moved it out of the else statement.